velox/type/DecimalUtil.h

```diff
@@ -173,11 +173,7 @@ class DecimalUtil {
     }
     // Check overflow.
     if (!valueInPrecisionRange(rescaledValue, toPrecision) || isOverflow) {
-      VELOX_USER_FAIL(
```

Comment:
Wait [#5307](https://github.com/facebookincubator/velox/pull/5307/files#diff-c6d2f99b68772f908f2f3e8b979b3c2894f790f20af68684e7b0400bed969389R156) merged.

Reply:
Can we remove the modification here and mark the PR as draft until 5307 merge?
Gluten has conflict between 6020 and 5307 which break the auto PR pick now. We have to manually resolve it each time

Reply:
@FelixYBW OK.

Reply:
@FelixYBW But there is one problem with this. The decimal avg function requires the use of the modified rescaleWithRoundUp method, which needs to be called with the "overflow" parameter for validation. Otherwise, there might be issues with Spark's unit testing. But if I pass these unmodified parameters in my PR, it will not compile successfully.

Reply:
Eventually the code should like below, right?

if (!valueInPrecisionRange(rescaledValue, toPrecision) || isOverflow) {
  if (throwIfOverFlow){
    Velox_USER_FAIL()
  } else{
    isOverflow = true;
    return return std::nullopt;    
    //after both 5307 and 6020 are merged, should we still return nullopt even isOverflow already pass the info to caller?
  }
}

It's OK the PR can't be compile since it waits until 5307 is merged

Reply:
@FelixYBW Done.